src/webgpu/api/validation/createTexture.spec.ts

```diff
@@ -70,45 +70,173 @@ class F extends ValidationTest {
-    { sampleCount: 0, _success: false }, // sampleCount of 0 is not allowed
-    { sampleCount: 1, _success: true }, // sampleCount of 1 is allowed
-    { sampleCount: 2, _success: false }, // sampleCount of 2 is not allowed
-    { sampleCount: 3, _success: false }, // sampleCount of 3 is not allowed
-    { sampleCount: 4, _success: true }, // sampleCount of 4 is allowed
-    { sampleCount: 8, _success: false }, // sampleCount of 8 is not allowed
-    { sampleCount: 16, _success: false }, // sampleCount of 16 is not allowed
-    { sampleCount: 4, mipLevelCount: 2, _success: false }, // multisampled multi-level not allowed
-    { sampleCount: 4, arrayLayerCount: 2, _success: false }, // multisampled multi-layer is not allowed
+g.test('zero_size')
+  .desc(
+    `Test texture creation with zero or nonzero size of
+    width, height, depthOrArrayLayers and mipLevelCount for every dimension, and representative formats.
+    TODO: add tests for depth/stencil format if depth/stencil format can support mipmaps.`
+  )
+  // .cases(poptions('dimension', ['1d', '2d', '3d'] as const))
+  .subcases(() =>
+    params()
+      .combine(poptions('dimension', ['1d', '2d', '3d'] as const))
+      .combine(
+        poptions('zeroArgument', [
+          'none',
+          'width',
+          'height',
+          'depthOrArrayLayers',
+          'mipLevelCount',
+        ] as const)
+      )
+      .combine(poptions('format', ['rgba8unorm', 'rgb10a2unorm', 'bc1-rgba-unorm'] as const))
+      .unless(({ format, dimension }) => format === 'bc1-rgba-unorm' && dimension !== '2d')
+  )
+  .fn(async t => {
+    const { dimension, zeroArgument, format } = t.params;
+
+    const size = dimension === '1d' ? [32, 1, 1] : dimension === '2d' ? [32, 32, 1] : [32, 32, 32];
+    let mipLevelCount = 1;
+
+    switch (zeroArgument) {
+      case 'width':
+        size[0] = 0;
+        break;
+      case 'height':
+        size[1] = 0;
+        break;
+      case 'depthOrArrayLayers':
+        size[2] = 0;
+        break;
+      case 'mipLevelCount':
+        mipLevelCount = 0;
+        break;
+      default:
+        break;
+    }
+
+    const descriptor = {
+      size,
+      mipLevelCount,
+      dimension,
+      format,
+      usage: GPUTextureUsage.SAMPLED,
+    };
+
+    await t.selectDeviceOrSkipTestCase(kAllTextureFormatInfo[format].extension);
+
+    const success = zeroArgument === 'none';
+    t.expectValidationError(() => {
+      t.device.createTexture(descriptor);
+    }, !success);
+  });
+
+g.test('mipLevelCount,format')
+  .desc(
+    `Test texture creation with no mipmap chain, partial mipmap chain, full mipmap chain, out-of-bounds mipmap chain
+    for every format with different texture dimension types.
+    TODO: test 1D and 3D dimensions. Note that it is invalid for some formats with 1D/3D and/or mipmapping.`
+  )
+  .subcases(() =>
+    params()
+      .combine(poptions('format', kAllTextureFormats))
```

Comment:
I noticed that a few texture formats are missing in CTS and Dawn, compared to spec.
The texture formats we are missing in CTS: depth16unorm, depth24unorm-stencil8, depth32float-stencil8
The texture formats we are missing in Dawn: stencil8 + all missing formats in CTS. 

Do we need to file bugs for this? I am not sure whether you are aware of this, though.

Reply:
Sorry, forgot about this. Yeah, we should add them in the CTS. Filed https://github.com/gpuweb/cts/issues/467